### PR TITLE
Feature/refactor args

### PR DESCRIPTION
**Pull request by CatarinaCarolina**
build input argument parsing and loading into relevant classes
test metagenomic contig AS output and make proto_cluster.category an optional attribute

### CODE REVIEW

File: src/file_input/load_files.py [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+# from other modules
 from src.genbank.gbk import GBK
 
 
-def load_datset_folder(path: Path) -> list[GBK]:
+def load_datset_folder(
+    path: Path,
+    mode: str = "recursive",
+    include_gbk: List[str] = ["cluster", "region"],
+    exclude_gbk: List[str] = ["final"],
```
[the remark this answers is not in the document]
Comment (adraismawur) [reply src/file_input/load_files.py:16]: ok it's an antipattern

set to none first, then set in method body

File: src/parameters/cmd_parser.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    """
+
+    parser = ArgumentParser(
+        prog="BiG-SCAPE",
+        description="Biosynthetic Gene Similarity Clustering and Prospecting Engine",
+        epilog="For a more comprehensive help menu and tutorials see GitHub Wiki",
+    )
+
+    parser.add_argument(
+        "--label",
```
Comment (adraismawur) [inline src/parameters/cmd_parser.py:25]: arguments to add_argument consistent order

File: src/parameters/diagnostics.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        quiet: bool
+        profiling: bool
+        log_level: int
+    """
+
+    def __init__(self) -> None:
+        self.verbose: Optional[bool] = None
+        self.quiet: Optional[bool] = None
+        self.profiling: Optional[bool] = None
+        self.log_level: Optional[int] = None
```
Comment (adraismawur) [inline src/parameters/diagnostics.py:27]: remove log_level

File: src/parameters/run.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
+"""Module containing a class for a BiG-SCAPE run object, which has all the
+run parameters/arguments"""
+
+# from python
+from typing import Optional, Any
+
+# from dependencies
+# from other modules
+# from this module
+from src.parameters.input import Input
```
Comment (adraismawur) [inline src/parameters/run.py:10]: rename all to xParameters e.g. InputParameters

File: src/parameters/hmmer.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+
+class Hmmer:
+    """
+    Class to store all run hmmer parameters
+
+    Attributes:
+        domain_overlap_cutoff: float
+        force_hmmscan: Bool
+        skip_alignment: Bool
```
Comment (adraismawur) [inline src/parameters/hmmer.py:23]: refactor to force_hmmalign

File: src/parameters/hmmer.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                    lines,
+                )
+
+                if not all(lines_valid):
+                    logging.error("Invalid Pfam accession(s)")
+                    raise InvalidInputArgError
+
+                self.domain_includelist = lines
+
+                # TODO: test this
```
Comment (adraismawur) [inline src/parameters/hmmer.py:82]: remove

**Review by adraismawur — CHANGES_REQUESTED** [review]

File: src/errors/input_args.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+"""Contains errors raised for input argument parsing"""
+
+
+class InvalidInputArgError(Exception):
+    """Raised when a validation error occurs in parsing a commandlind input
+    argument"""
+
+    def __init__(self):
+        super().__init__(
+            "Validation error occurred when parsing a command\
```
[the remark this answers is not in the document]
Comment (CatarinaCarolina) [reply src/errors/input_args.py:10]: refactored

**Review by adraismawur — APPROVED** [review]

**Pull request merged**